packages/product/src/services/product-option.ts

```diff
@@ -18,7 +30,59 @@ export default class ProductOptionService<
       productOptionRepository as ProductOptionRepository
   }
 
-  @InjectTransactionManager(doNotForceTransaction, "productOptionRepository_")
+  async retrieve(
+    productOptionId: string,
+    config: FindConfig<ProductTypes.ProductOptionDTO> = {},
+    @MedusaContext() sharedContext?: Context
+  ): Promise<TEntity> {
+    return (await retrieveEntity<
+      ProductOption,
+      ProductTypes.ProductOptionDTO
+    >({
+      id: productOptionId,
+      entityName: ProductOption.name,
+      repository: this.productOptionRepository_,
+      config,
+      sharedContext,
+    })) as TEntity
+  }
+
+  async list(
+    filters: ProductTypes.FilterableProductOptionProps = {},
+    config: FindConfig<ProductTypes.ProductOptionDTO> = {},
+    @MedusaContext() sharedContext?: Context
+  ): Promise<TEntity[]> {
+    return (await this.productOptionRepository_.find(
+      this.buildQueryForList(filters, config),
+      sharedContext
+    )) as TEntity[]
+  }
+
+  async listAndCount(
+    filters: ProductTypes.FilterableProductOptionProps = {},
+    config: FindConfig<ProductTypes.ProductOptionDTO> = {},
+    @MedusaContext() sharedContext?: Context
+  ): Promise<[TEntity[], number]> {
+    return (await this.productOptionRepository_.findAndCount(
+      this.buildQueryForList(filters, config),
+      sharedContext
+    )) as [TEntity[], number]
+  }
+
+  private buildQueryForList(
+    filters: ProductTypes.FilterableProductOptionProps = {},
+    config: FindConfig<ProductTypes.ProductOptionDTO> = {},
+  ) {
+    const queryOptions = ModulesSdkUtils.buildQuery<ProductOption>(filters, config)
+
+    if (filters.title) {
+      queryOptions.where["title"] = { $ilike: filters.title }
+    }
+
+    return queryOptions
+  }
+
+  @InjectTransactionManager(shouldForceTransaction, "productOptionRepository_")
```

Comment:
comment to all the methods of all the services except `product-module-service.ts`:
I believe we don't need to inject the context here as they will come from the main service of the module.

Reply:
Should we also add `@InjectManager` on all the read methods? retrieve, list, etc?

Reply:
We need this injection in both places because we test these services independently. Additionally, these services are also "accessible" by the user directly, so incase that happens, its better for it to be injected in both places. There's no harm in it as they reuse a manager if it exists.

Reply:
But in this case, if the module is running in isolated mode, `shouldForceTransaction` will always be true and a new transaction will always be initialized.

Reply:
Even in that case, if the context already has a manager, it reuses it. Take a look here - https://github.com/medusajs/medusa/blob/feat/tags-and-types/packages/product/src/repositories/base.ts#L25-L28

Let me know if i missed anything @carlos-r-l-rodrigues

Reply:
oh my bad. `transaction` is a method of the class, not the ORM directly.
Thanks